src/hotspot/share/gc/shared/weakProcessor.hpp

```diff
@@ -71,6 +71,7 @@ class WeakProcessor : AllStatic {
   class Task;
 
 private:
+  template<typename IsAlive, typename KeepAlive> class CountingClosure;
```

Comment:
May I suggest breaking this into two lines? It's too easy to start to read this as:
```
template<typename ...>
class GangTask;
```

Reply:
Done.